ui/apps/platform/src/Containers/NetworkGraph/TopologyComponent.tsx

```diff
@@ -110,13 +126,15 @@ const TopologyComponent = ({
         } else if (newSelectedEntity) {
             setDefaultDeploymentTab(deploymentTabs.DETAILS);
             const { data, id } = newSelectedEntity;
-            const [newDetailType, newDetailId] = getUrlParamsForEntity(data.type, id);
+            const [newNodeType, newNodeId] = getUrlParamsForNode(data.type, id);
             const queryString = clearSimulationQuery(history.location.search);
             // if found, and it's not the logical grouping of all external sources, then trigger URL update
-            if (newDetailId !== 'EXTERNAL') {
-                const newURL = `${networkBasePath}/${newDetailType}/${encodeURIComponent(
-                    newDetailId
-                )}${queryString}`;
+            if (newNodeId !== 'EXTERNAL') {
+                let newURL = `${networkBasePath}/${newNodeType}/${encodeURIComponent(newNodeId)}`;
+                if (externalIP) {
+                    newURL = `${newURL}/externalIP/${externalIP}`;
+                }
```

Comment:
We may need to rethink how we construct the paths if we deal with other nested types.